### ISSUE TITLE
You do not have permission to add a plugin on edit post page

### ISSUE THREAD

**Issue by suquant** [nephila/djangocms-blog#162]
I installed django-cms dev (https://github.com/divio/django-cms/commit/c5523556c355f504a6f8068b8a7b4449effba6b6) (the latest version as of today)
djangocms-blog dev (https://github.com/nephila/djangocms-blog/commit/ec423a8d6870c846f46c0fcc805957e0a1309d04) (the latest version as of today)
 on top of an existing Django 1.8.5 / Python 2.7.10 project with postgresql as the db.

Having created a simple template and placeholder, I get an error when trying to add a plugin using the front end content editor (for example the "Generic / Picture" plugin). Here's my template file:
```
{% extends "magazine/base.html" %}
{% load cms_tags %}

{% block base_content %}
  {% placeholder template_1_content %}
{% endblock %}
```

Even though I am logged in as superuser a pop-up appears saying "You do not have permissions to add a plugin" when I attempt to add a plugin.

Aside from not being able to add plugins, everything else about the django-cms interface seems to be running fine - I can view the admin, create pages, toggle between live and draft, etc.

I have looked inside the django-cms code and it appears the error is originating from a publisher_is_draft check inside the class cms.admin.pageadmin.PageAdmin:

```
def has_add_plugin_permission(self, request, placeholder, plugin_type):
        if not permissions.has_plugin_permission(request.user, plugin_type, "add"):
            return False
        page = placeholder.page
        if page and not page.has_change_permission(request):
            return False
        if page and not page.publisher_is_draft:
            return False
        return True
```
Basically, it appears django-cms does not think the page is in draft mode, even though I have entered 'draft mode' according to the front end (ie: toggled the 'live' mode icon to 'draft'). The fact that the other checks in the above function pass confirms that the error is not a user permission issue (indeed, as I mentioned above, the error happens as superuser) but something about the page state.

Also when i edit blog post on the url, for example "/en/2015/11/08/my-second-post/" the "page" in "has_add_plugin_permission" function is "/en/" which is the main page connected to BlogApplication in advanced settings.

Really i do not understand how possible edit blog post by frontend editor.
Any ideas?

**Comment by yakky** [comment]
Is the template a page template or post_detail.HTML template? Bear in mind that you can't use placeholder in templates which are rendered by applications, you must use static_template for that. If you attach the blog application in the root of the website, you can't use 'placeholder' in any page, and depending in the URL layout you can't even use pages but only posts. Could you provide more information on the structure of your templates?

**Comment by suquant** [comment]
@yakky 
> If you attach the blog application in the root of the website, you can't use 'placeholder' in any page, and depending in the URL layout you can't even use pages but only posts

Yes, i understand it, it for example, i tested for /en/magazine/ page also but same errors was happen.

settings.py:

```
INSTALLED_APPS += [
    'easy_thumbnails',
    'parler',
    'reversion',
    'sortedm2m',
    'taggit',
    'taggit_autosuggest',
    'meta',
    'meta_mixin',
    'treebeard',
    'sekizai',
    'menus',
    'djangocms_admin_style',
    'mptt',

    'cms',
    'djangocms_blog',
    'djangocms_text_ckeditor',

    'djangocms_style',
    'djangocms_column',
    'djangocms_grid',
    'djangocms_oembed',
    'djangocms_file',
    'djangocms_picture',
    'djangocms_teaser',
    'djangocms_video',
    'djangocms_googlemap',
    'djangocms_inherit',
    'djangocms_link',
    'djangocms_snippet',
    'djangocms_table',

    'cmsplugin_filer_link',
    'cmsplugin_filer_folder',
    'cmsplugin_filer_file',
    'cmsplugin_filer_image',
    'cmsplugin_filer_teaser',
    'cmsplugin_filer_video',
    'filer',
]

THUMBNAIL_PROCESSORS = (
    'easy_thumbnails.processors.colorspace',
    'easy_thumbnails.processors.autocrop',
    # 'easy_thumbnails.processors.scale_and_crop',
    'filer.thumbnail_processors.scale_and_crop_with_subject_location',
    'easy_thumbnails.processors.filters',
    'easy_thumbnails.processors.background',
)

META_SITE_PROTOCOL = 'http'
META_USE_SITES = True

MIDDLEWARE_CLASSES += [
    'cms.middleware.user.CurrentUserMiddleware',
    'cms.middleware.page.CurrentPageMiddleware',
    'cms.middleware.toolbar.ToolbarMiddleware',
    'cms.middleware.language.LanguageCookieMiddleware',
]

TEMPLATES[0]['OPTIONS']['context_processors'] += [
    'sekizai.context_processors.sekizai',
    'cms.context_processors.cms_settings',
]



CMS_TEMPLATES = (
    ('magazine/template_1.html', 'Template One'),

)

BLOG_PERMALINK_URLS = {
    'full_date': r'^(?P<year>\d{4})/(?P<month>\d{1,2})/(?P<day>\d{1,2})/(?P<slug>\w[-\w]*)/$',
    'short_date': r'^(?P<year>\d{4})/(?P<month>\d{1,2})/(?P<slug>\w[-\w]*)/$',
    'category': r'^post/(?P<category>\w[-\w]*)/(?P<slug>\w[-\w]*)/$',
    'slug': r'^post/(?P<slug>\w[-\w]*)/$',
}

TEXT_SAVE_IMAGE_FUNCTION = 'djangocms_text_ckeditor.picture_save.create_picture_plugin'
CKEDITOR_SETTINGS = {
    'language': '{{ language }}',
    'toolbar': 'CMS',
    'skin': 'moono',
}
```

template_1.html:

```
{% extends "magazine/base.html" %}
{% load cms_tags %}

{% block base_content %}
  {% placeholder template_1_content %}
{% endblock %}
```
magazine/base.html:

```
{% load cms_tags sekizai_tags %}
<html>
  <head>
      <title>{% page_attribute "page_title" %}</title>
      {% render_block "css" %}
  </head>
  <body>
      {% cms_toolbar %}
      {% placeholder base_content %}
      {% block base_content %}{% endblock %}
      {% render_block "js" %}
  </body>
</html>
```

**Comment by suquant** [comment]
@yakky do you have some ideas ?

**Comment by yakky** [comment]
The template which is applied to the home page and root of the blog post does not a **content** ``block`` the django_blog base template can extend, the when you open the blog post detail page the standard page template is used (**magazine/template_1.html**) which contains only ``placeholder`` which are not editable when in a apphooked view. Templates should look something like https://gist.github.com/yakky/20df5994f04ad3f31b73

**Comment by suquant** [comment]
@yakky Yes this solution fix my problem :) Thank you.
